ui/lib/core/addon/components/text-file.hbs

```diff
@@ -1,62 +1,54 @@
-{{#unless this.inputOnly}}
+{{#unless @hideTextAreaToggle}}
   <div class="level is-mobile">
     <div class="level-left">
-      <label class="is-label" data-test-text-label={{true}}>
-        {{#if this.label}}
-          {{this.label}}
-          {{#if @helpText}}
-            <InfoTooltip>
-              <span data-test-help-text>
-                {{@helpText}}
-              </span>
-            </InfoTooltip>
-          {{/if}}
-        {{else}}
-          File
+      <label class="is-label" data-test-text-file-label>
+        {{or @label "File"}}
+        {{#if @helpText}}
+          <InfoTooltip>
+            <span data-test-help-text>
+              {{@helpText}}
+            </span>
+          </InfoTooltip>
         {{/if}}
       </label>
     </div>
     <div class="level-right">
       <div class="control is-flex">
-        <input
+        <Input
           data-test-text-toggle
-          id={{concat "useText-" this.elementId}}
-          type="checkbox"
-          name={{concat "useText-" this.elementId}}
+          id="text-area-toggle"
           class="switch is-rounded is-success is-small"
-          checked={{@file.enterAsText}}
-          {{on "change" (toggle "enterAsText" @file)}}
+          @type="checkbox"
+          @checked={{this.showTextInput}}
+          {{on "change" (fn (mut this.showTextInput) (not this.showTextInput))}}
         />
-        <label for={{concat "useText-" this.elementId}}>
+        <label for="text-area-toggle">
           Enter as text
         </label>
       </div>
     </div>
   </div>
 {{/unless}}
-<div
-  class="field text-file box is-fullwidth is-marginless is-shadowless {{if this.inputOnly 'is-paddingless'}}"
-  data-test-component="text-file"
->
-  {{#if @file.enterAsText}}
+<div class="field text-file box is-fullwidth is-marginless is-shadowless is-paddingless" data-test-component="text-file">
+  {{#if this.showTextInput}}
     <div class="control has-icon-right">
       <textarea
         class="textarea {{unless this.showValue 'masked-font'}}"
-        {{on "input" this.updateData}}
-        data-test-text-file-textarea={{true}}
-      >{{@file.value}}</textarea>
+        {{on "input" this.handleTextInput}}
+        data-test-text-file-textarea
+      >
+        {{@file.value}}
+      </textarea>
       <button
         {{on "click" this.toggleMask}}
         type="button"
-        class="{{if (eq this.value '') 'has-text-grey'}} masked-input-toggle button {{if this.displayOnly 'is-compact'}}"
-        data-test-button
+        class="{{if (eq this.value '') 'has-text-grey'}} masked-input-toggle button is-compact"
+        data-test-button="toggle-masked"
       >
-        <Icon @name={{if this.showValue "eye" "eye-off"}} />
+        <Icon @name={{if this.showValue "eye" "eye-off"}} @stretched={{true}} />
```

Comment:
before this arg was added 😂 
<img width="355" alt="Screen Shot 2022-12-16 at 4 47 55 PM" src="https://user-images.githubusercontent.com/68122737/208214220-b6953598-3a15-4e52-b4c4-4917ab21bb44.png">

after
<img width="212" alt="Screen Shot 2022-12-16 at 4 48 25 PM" src="https://user-images.githubusercontent.com/68122737/208214246-4b187e88-a537-4fcb-9eb5-c928ecd3b7ef.png">